tracer.go

```diff
@@ -121,6 +125,110 @@ type StartSpanOptions struct {
 	// identical to those for Span.SetTag(). May be nil.
 	//
 	// If specified, the caller hands off ownership of Tags at
-	// StartSpanWithOptions() invocation time.
+	// StartSpan() invocation time.
 	Tags map[string]interface{}
 }
+
+// StartSpanOption instances (zero or more) may be passed to Tracer.StartSpan.
+type StartSpanOption interface {
+	Apply(*StartSpanOptions)
+}
+
+// ReferenceType is an enum type describing different categories of
+// relationships between two Spans. If Span-2 refers to Span-1, the
+// ReferenceType describes Span-1 from Span-2's perspective. For example,
+// ParentRef means that Span-1 created Span-2.
+//
+// NOTE: Span-1 and Span-2 do *not* necessarily depend on each other for
+// completion; e.g., Span-2 may be part of a background job enqueued by Span-1,
+// or Span-2 may be sitting in a distributed queue behind Span-1.
+type ReferenceType int
+
+const (
+	// ParentRef refers to a "parent" Span that created the new Span. The
+	// expectation is that the parent Span has not finished by the time the new
+	// Span starts. Though it is often the case that the parent Span cannot
+	// complete successfully until the new Span does as well, this is not a
+	// semantic requirement for ParentRef.
+	//
+	// A timing diagram for a ParentRef that's blocked on the new Span:
+	//
+	//     [-ParentRef Span------]
+	//          [-New Span------]
+	//
+	// A timing diagram for a ParentRef that's not blocked on the new Span:
+	//
+	//     [-ParentRef Span-]
+	//                  [-New Span-]
+	//
+	// See opentracing.Parent()
+	ParentRef ReferenceType = iota
```

Comment:
Eventually this would go to the Spec, but for now I think it's worth clarifying a few semantic points:

1. If there is no ParentRef in the StartSpanOptions, a new trace / root span is created
1. ParentRef may be accompanied by other Refs, e.g. below Span C can be started with Parent(A) and Predecessor(B)

```
[--Span A------------------------]
   [--Span B--]  [--Span C--]
```